File: scripts/gc_translation/glossary.py
"""Load and format the theological glossary for prompt injection."""
from __future__ import annotations

from pathlib import Path

import yaml

from scripts.gc_translation.paths import GLOSSARY_PATH


def load_glossary(path: Path = GLOSSARY_PATH) -> dict[str, str]:
    raw = yaml.safe_load(Path(path).read_text(encoding="utf-8"))
    flat: dict[str, str] = {}
    for _group, entries in raw.items():
        flat.update(entries)
    return flat


def format_glossary_for_prompt(glossary: dict[str, str]) -> str:
    lines = ["| English | Vietnamese |", "|---------|------------|"]
    for en, vi in sorted(glossary.items(), key=lambda kv: kv[0].lower()):
        lines.append(f"| {en} | {vi} |")
    return "\n".join(lines)

File: scripts/egw_translate/lib/glossary.ts
import { parseYaml } from "./yaml";

export type Glossary = Map<string, string>;

export async function loadGlossary(path: string): Promise<Glossary> {
  const text = await Bun.file(path).text();
  const parsed = parseYaml(text);
  const out = new Map<string, string>();
  for (const v of Object.values(parsed)) {
    if (typeof v === "string") continue;
    for (const [en, vi] of Object.entries(v)) {
      out.set(en, vi);
    }
  }
  return out;
}

export function formatGlossaryTable(g: Glossary): string {
  const rows: string[] = ["| English | Vietnamese |", "| --- | --- |"];
  for (const [en, vi] of g) {
    rows.push(`| ${en} | ${vi} |`);
  }
  return rows.join("\n");
}
